swifttool/manager.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright (c) 2014, Blue Box Group, Inc.
+# Copyright (c) 2014, Craig Tracey <craigtracey@gmail.com>
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+#
+
+import copy
+import logging
+import math
+import os
+import shutil
+import time
+import yaml
+
+from swifttool.ring_defintion import ringsdef_helper
+from swift.common.ring import RingBuilder
+from swifttool.utils import RING_TYPES, get_devices
+
+LOG = logging.getLogger(__name__)
+
+_STATUS_FILE = 'capacity.status'
+
+
+class CapacityManager(object):
+
+    def __init__(self, iterations=10,
+                 def_file='/etc/swift/ring_definition.yml',
+                 scaleup=True, metadata=None,
+                 outdir='/etc/swift/'):
+        self.iterations = int(iterations)
+        self.def_file = def_file
+        self.scaleup = scaleup
+        self.cluster_changes = {}
+        self.outdir = outdir
+        self.cluster_config = None
+        self.metadata = metadata
+        # Save the copy
+        shutil.copyfile(self.def_file, self.def_file + ".orig")
```

Comment:
I tend to keep "work" out of the init function - the ramification here is merely _instantiating_ this class has a side effect of copying files around the filesystem.  I'd really keep out of the business of doing copies from `__init__`